### PR TITLE
update documentation in protocol.d.ts

### LINKED ISSUE #11486: 2.0.6 TSServer protocol.d.ts: new features are missing documentation
**TypeScript Version:**  2.0.6-Insider

All new features are missing documentation. Hard to use without knowing how to :-)

### PR DESCRIPTION
**Pull request by vladima**
fixes #11486

NOTE: seems that we lack response types for majority of new messages, I'll add them in the follow-up PR.

### CODE REVIEW

File: src/server/protocol.d.ts [hunk 1]
```diff
@@ -503,17 +575,47 @@ declare namespace ts.server.protocol {
...
+         */
         hasMixedContent?: boolean;
+        /**
+         * Content of the file
+         */
         content?: string;
     }
 
+    /**
+     * Represent an external project
```
Comment (mhegazy) [inline src/server/protocol.d.ts]: can you elaborate on what an `external` project is., and how it works, and the expectations in cases of having multiple tsconfig.json in it, etc..

File: src/server/protocol.d.ts [hunk 2]
```diff
@@ -689,55 +842,112 @@ declare namespace ts.server.protocol {
...
+     * Request to set compiler options for inferred projects. 
+     */
     export interface SetCompilerOptionsForInferredProjectsRequest extends Request {
         arguments: SetCompilerOptionsForInferredProjectsArgs;
     }
 
     /**
+     * Argument for SetCompilerOptionsForInferredProjectsRequest request.
+     */
+    export interface SetCompilerOptionsForInferredProjectsArgs {
```
Comment (mhegazy) [inline src/server/protocol.d.ts]: we should elaborate on what "InferredProjects" are, and how they are different from "external" projects, and normal projects.

File: src/server/protocol.d.ts [hunk 3]
```diff
@@ -754,23 +964,47 @@ declare namespace ts.server.protocol {
     export interface CloseRequest extends FileRequest {
     }
 
+    /**
+     * Request to obtain the list of files that should be regenerated if target file is recompiled.
```
Comment (mhegazy) [inline src/server/protocol.d.ts]: I would add a note, that this does not generate any outputs. and a reference to `CompileOnSaveEmitFileRequest` for generation.,

File: src/server/protocol.d.ts [hunk 4]
```diff
@@ -754,23 +964,47 @@ declare namespace ts.server.protocol {
...
+     * Response for CompileOnSaveAffectedFileListRequest request; 
+     */
     export interface CompileOnSaveAffectedFileListResponse extends Response {
         body: CompileOnSaveAffectedFileListSingleProject[];
     }
 
+    /**
+     * Request to recompile the file.
+     */
     export interface CompileOnSaveEmitFileRequest extends FileRequest {
```
Comment (mhegazy) [inline src/server/protocol.d.ts]: i would make it clear that this actually writes files to disk.

File: src/server/protocol.d.ts (hunk 1 above)
Comment (vladima) [inline src/server/protocol.d.ts]: ✔️

File: src/server/protocol.d.ts (hunk 3 above)
Comment (vladima) [inline src/server/protocol.d.ts]: :heavy_check_mark:

File: src/server/protocol.d.ts (hunk 4 above)
Comment (vladima) [inline src/server/protocol.d.ts]: :heavy_check_mark:

File: src/server/protocol.d.ts [hunk 5]
```diff
@@ -966,6 +978,7 @@ declare namespace ts.server.protocol {
 
     /**
      * Request to obtain the list of files that should be regenerated if target file is recompiled.
+     * NOTE: this us query only operation and does not generate any output on disk.
```
Comment (mhegazy) [inline src/server/protocol.d.ts]: type: `this is a query-only operation`

Comment (vladima) [inline src/server/protocol.d.ts]: ✔️

**Comment by mhegazy** [comment]
@vladima can you port this to release-2.0.5 branch

**Pull request merged**